tests/src/Kernel/GraphQLTestBase.php

```diff
@@ -79,6 +82,26 @@ protected function setUp(): void {
     $this->builder = new ResolverBuilder();
   }
 
+  /**
+   * Configures core's cache policy.
+   *
+   * Modifies the DefaultRequestPolicy classes, which always add in the
+   * CommandLineOrUnsafeMethod policy which will always result in DENY in a
+   * Kernel test because we're running via the command line.
+   *
+   * @param int $max_age
+   *   Max age to cache responses.
+   */
+  protected function configureCachePolicy(int $max_age = 900) {
+    $this->container->set('dynamic_page_cache_request_policy', (new ChainRequestPolicy())
+      ->addPolicy(new DenyPost()));
+    $this->container->set('page_cache_request_policy', (new ChainRequestPolicy())
+      ->addPolicy(new NoSessionOpen($this->container->get('session_configuration')))
+      ->addPolicy(new DenyPost()));
+    // Turn on caching.
+    $this->config('system.performance')->set('cache.page.max_age', $max_age)->save();
+  }
```

Comment:
Is this needed in the test? Kernel tests install the `graphql` module and it should load the services for core and the module on its own. I believe configuring this in the test bypasses actually testing our configuration (i.e. removing the module's service definition could still let the test pass even though we may not want it to).

Reply:
yes, it is needed - as per the comment core adds in the CommandLineOrUnsafeMethod by default which means in the test it is always DENY because kernel tests are command line based

All this is doing is recreating those services, but without CommandLineOrUnsafeMethod

Reply:
Alternatively, I could write these tests as a functional test, but I didn't know if there was a reason the module didn't have any yet, so didn't want to introduce one without checking

Reply:
That's what I get for doing reviews at 8 AM without coffee. Totally missing a wonderfully written comment.

Reply:
❤️